packages/plugins/mocha-tests/src/lib/index.js

```diff
@@ -83,82 +106,79 @@ class MochaTestRunner {
-      },
-      (provider, next) => { // set provider
-        web3 = new Web3(provider);
-        next();
-      },
-      (next) => { // get accounts
-        web3.eth.getAccounts((err, accts) => {
-          if (err) {
-            return next(err);
-          }
+    const provider = await this.events.request2("tests:blockchain:start", this.options);
+    this.web3 = new Web3(provider);
+    accounts = await this.web3.eth.getAccounts();
+    await events.request2("contracts:reset");
+    let contractFiles = await events.request2("config:contractsFiles");
 
-          accounts = accts;
-          next();
-        });
-      },
-      (next) => { // reset contracts as we might have state leakage from other plugins
-        events.request("contracts:reset", next);
-      },
-      (next) => { // get contract files
-        events.request("config:contractsFiles", next);
-      },
-      (cf, next) => {
-        plugins.emitAndRunActionsForEvent('tests:contracts:compile:before', cf, next);
+    async.waterfall([
+      (next) => {
+        this.plugins.emitAndRunActionsForEvent('tests:contracts:compile:before', contractFiles, next);
       },
-      (cf, next) => { // compile contracts
-        events.request("compiler:contracts:compile", cf, next);
+      (_contractFiles, next) => {
+        contractFiles = _contractFiles;
+        events.request("compiler:contracts:compile", _contractFiles, next);
       },
-      (cc, next) => {
-        plugins.emitAndRunActionsForEvent('tests:contracts:compile:after', cc, next);
+      (_compiledContracts, next) => {
+        this.plugins.emitAndRunActionsForEvent('tests:contracts:compile:after', _compiledContracts, next);
       },
-      (cc, next) => { // override require
-        compiledContracts = cc;
-
-        Module.prototype.require = function(req) {
-          const prefix = "Embark/contracts/";
-          if (!req.startsWith(prefix)) {
-            return originalRequire.apply(this, arguments);
-          }
-
-          const contractClass = req.replace(prefix, "");
-          const instance = compiledContracts[contractClass];
-
-          if (!instance) {
-            throw new Error(`Cannot find module '${req}'`);
-          }
-
-          return instance;
-        };
-        next();
-      },
-      (next) => { // initialize Mocha
-        const mocha = new Mocha();
-
-        mocha.reporter(Reporter, { reporter: reporter });
-        const describeWithAccounts = (scenario, cb) => {
-          Mocha.describe(scenario, cb.bind(mocha, accounts));
-        };
-
-        mocha.suite.on('pre-require', () => {
-          global.describe = describeWithAccounts;
-          global.contract = describeWithAccounts;
-          global.assert = assert;
-          global.config = config;
+      (_compiledContracts, next) => {
+      compiledContracts = _compiledContracts;
+        const fns = this.files.map((file) => {
+          return (seriesCb) => {
+
+            const mocha = new Mocha();
```

Comment:
We're still negating the potential performance gains of a shared mocha instance here. Is there a reason why this is being moved?